src/contracts/core/RewardsCoordinator.sol

```diff
@@ -271,6 +289,69 @@ contract RewardsCoordinator is
         _setGlobalOperatorCommission(_globalCommissionBips);
     }
 
+    /// @inheritdoc IRewardsCoordinator
+    function setOperatorAVSCommission(address operator, address avs, uint16 commission) external {
+        require(msg.sender == operator, "RewardsCoordinator.setOperatorAVSCommission: caller is not the operator");
+        require(
+            commission <= ONE_HUNDRED_IN_BIPS,
+            "RewardsCoordinator.setOperatorAVSCommission: commission must be <= 10000 bips"
+        );
+        uint32 activatedAt = SafeCast.toUint32(block.timestamp) + activationDelay;
+
+        OperatorCommission storage operatorCommission = operatorAVSCommissionBips[operator][avs];
+
+        // If, the earlier 'new' commission is activated, we update the 'old' commission with the earlier 'new' commission.
+        // Else, the earlier 'old' commission remains the same. This is essentially resetting the activation delay window
+        // since the earlier commission setting didn't complete.
+        if (block.timestamp >= operatorCommission.activatedAt) {
+            operatorCommission.oldCommissionBips = operatorCommission.newCommissionBips;
+        }
+        operatorCommission.newCommissionBips = commission;
+        operatorCommission.activatedAt = activatedAt;
```

Comment:
Happy to clarify if this section is confusing. It's essentially handling the case if an operator tries setting a commission again before their previous commission activates. The intended behavior here in that case is that the activation delay window is just extended starting from the current block.timestamp.